Catalog/src/Application/Product/Commands/DeleteProduct/DeleteProductCommand.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Catalog.Application.Common.Exceptions;
+using Catalog.Application.Common.Interfaces;
+using Catalog.Application.Outbox;
+using MediatR;
+using Messaging.Contracts;
+
+namespace Catalog.Application.Products.Commands.DeleteProduct;
+
+public record DeleteProductCommand : IRequest<int>
+{
+    public int Id { get; init; }
+}
+
+public class DeleteProductCommandHandler : IRequestHandler<DeleteProductCommand, int>
+{
+    private readonly IApplicationDbContext _context;
+
+    public DeleteProductCommandHandler(IApplicationDbContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<int> Handle(DeleteProductCommand request, CancellationToken cancellationToken)
+    {
+        var entity = _context.Products.FirstOrDefault(c => c.Id == request.Id);
+        if (entity == null)
+            throw new NotFoundException();
+
+        try
+        {
+            await _context.StartTransaction();
```

Comment:
I believe the default isolation level for EF core is READ COMMITTED. In case of concurrent updates, locks should be acquired for the rows which is going to be updated.